app/main/views/questionnaire.py

```diff
@@ -49,6 +49,18 @@ def check_survey_state():
 @questionnaire_blueprint.after_request
 def add_cache_control(response):
     response.cache_control.no_cache = True
+
+    return response
+
+
+@questionnaire_blueprint.after_request
+def save_questionnaire_store(response):
+    if not current_user.is_anonymous:
```

Comment:
Can't get this approach to work - possibly different to the check_survey_state example because it's after the request?